### ISSUE TITLE
[Feature Request] Prevent zPIV change addresses from sending transparent transactions without consent

### ISSUE THREAD

**Issue by Rock-N-Troll** [PIVX-Project/PIVX#444]
Anonymity can be weakened if a user tries spending the change (PIVX) of a zPIV spend. If a change address is used to send PIVX, the zPIV spend address can now be linked to a chain of transactions from the change address (the change address and zPIV spend are forever linked in the block they were "generated in" on the blockchain). 

There is currently a feature to burn any change left over from a zPIV spend which guarantees a user will not give up anonymity through change address misuse. The argument that a user would rather burn the change instead of control it isn't practical in an optimal, real world, user-minded cryptocurrency. A better approach is to assume a user doesn't want to throw away change after every private transaction, especially if it isn't required.

I propose a validation rule be implemented to prevent a user from accidentally giving up his or her own privacy AND to enable that same user to later spend from change addresses when it becomes feasible with smaller zPIV denominations OR when more zPIV change UTXOs are generated that can be used to mint a new zPIV:

1. If an address is the change address of a zPIV spend (because it would be a brand new address, there is no linkage at this point), the user's wallet should not allow the user to spend any coins from that address unless the user explicitly knows he or she will be compromising his or her own privacy. 
This can be accomplished by alerting and very clearly warning the user whenever a transaction spend would consume 1 or more zPIV change UTXOs.

The first alternative a user has to not burning a zPIV transaction's change is the following: At some point in the future when smaller zPIV denominations are possible, these solo change addresses will become usable again without compromising privacy (because they can now be used to mint, for example, multiple 0.1 zPIVs from an older 0.99 zPIV spend change address). 

The second alternative a user has to not burning a zPIV transaction's change is the following: A user can combine multiple zPIV change UTXOs together to create a new zPIV when possible. Note that when using this approach, there is now a linkage created between zPIV spends used to create those change UTXOs. However, the worst case scenario of minting a zPIV from these change addresses is that one can only infer that some anonymous user M has spent some N amount of zPIV (to create N UTXOs now being combined to produce 1 zPIV). In this scenario, there is still no way to know who did so and as a result, privacy is still maintained.

In conclusion, it seems like preventing or warning a user from spending PIVX from zPIV change addresses is probably a safer approach to keeping the whole network more private.

**Comment by rejectedpromise** [comment]
moving to project -> https://github.com/PIVX-Project/PIVX/projects/23
